lib/src/main/java/uk/gov/di/ipv/core/library/validation/TokenRequestValidator.java

```diff
@@ -1,37 +1,118 @@
 
-            if (claimsSet != null) {
-                return ValidationResult.createValidResult();
+                if (clientAuthenticationMethod.equals(JWT)) {
+                    LOGGER.error("Missing client_assertion jwt for configured client {}", clientId);
+                    throw new ClientAuthenticationException(
+                            String.format(
+                                    "Missing client_assertion jwt for configured client '%s'",
+                                    clientId));
+                }
             }
-        } catch (ParseException e) {
-            LOGGER.error("Unable to parse Claims set {} ", e.getMessage());
-            return new ValidationResult<>(false, OAuth2Error.INVALID_CLIENT);
+        } else {
+            PrivateKeyJWT clientJwt;
+            try {
+                clientJwt = PrivateKeyJWT.parse(requestBody);
+
+                clientId = clientJwt.getClientID().getValue();
+
+                String clientAuthenticationMethod = getClientAuthenticationMethod(clientId);
+
+                if (clientAuthenticationMethod.equals(NONE)) {
+                    return;
+                }
+
+                verifier.verify(clientJwt, null, null);
+                validateExpiration(clientJwt.getJWTAuthenticationClaimsSet());
+            } catch (ParseException | InvalidClientException | JOSEException e) {
+                LOGGER.error("Validation of client_assertion jwt failed");
+                throw new ClientAuthenticationException(e);
+            }
+        }
+    }
+
+    private String getClientAuthenticationMethod(String clientId)
+            throws ClientAuthenticationException {
+        String clientAuthenticationMethod =
+                configurationService.getClientAuthenticationMethod(clientId);
+        if (clientAuthenticationMethod == null) {
+            LOGGER.error("Failed to find config for client ID: {}", clientId);
+            throw new ClientAuthenticationException(
+                    String.format("Config for client ID '%s' not found", clientId));
         }
-        return ValidationResult.createValidResult();
+        return clientAuthenticationMethod;
+    }
+
+    private void validateExpiration(JWTAuthenticationClaimsSet claimsSet)
+            throws InvalidClientException {
+        Date expirationTime = claimsSet.getExpirationTime();
+        String maxAllowedTtl = configurationService.getClientTokenTtl(clientId);
+
+        OffsetDateTime offsetDateTime =
+                OffsetDateTime.now().plusSeconds(Long.parseLong(maxAllowedTtl));
+        if (expirationTime.getTime() / 1000L > offsetDateTime.toEpochSecond()) {
+            LOGGER.error("Client JWT expiry date is too far in the future");
+            throw new InvalidClientException(
+                    "The client JWT expiry date has surpassed the maximum allowed ttl value");
+        }
+    }
+
+    public static ClientAuthenticationVerifier<Object> getClientAuthVerifier(
```

Comment:
As this is being called in the constructor I don't think it needs to be static. I think it was originally intended to be called externally and the return value passed into the constructor to allow injection. As we're not doing that I reckon just make it an instant method.

Reply:
change to concrete